### PR TITLE
use correct registry for a given scope

### PR DESCRIPTION
**Pull request by gustavnikolaj**
See #94

### CODE REVIEW

File: lib/npm.js [hunk 1]
```diff
@@ -148,8 +153,16 @@ NPMLocation.prototype = {
...
     })
     .then(function() {
+      var scope;
+      if (repo[0] === '@') {
+        var scopeMatch = repo.match(/^(@.*)\//);
+        if (scopeMatch[1]) {
+          scope = scopeMatch[1];
+        }
+      }
+      console.log(scope, self.registryURL(scope))
```
Comment (guybedford) [inline lib/npm.js]: console.log

**Comment by guybedford** [comment]
This is excellent. I don't have a private setup to test on here, but will gladly merge the PR on faith.

**Comment by gustavnikolaj** [comment]
I have tested it locally - in fact, I've been using it since I submitted the pull request, by manually linking it in :-) I had a dependency on a module that I couldn't publish to the public registry. I'm in good faith about this, but on the other hand, my experience with jspm is rather limited, so I don't know what the scope of my change is. 

Is the code that I have touched being called by any thing external to this module directly?

I think that you'd be able to test it, by trying to install a scoped package available in the public registry, and defining another registry for that scope. I have a package called `@gustavnikolaj/migrate` which is a fork of the migrate package with an unmerged PR of mine. If you put the following in `.npmrc`:

```
registry=https://registry.npmjs.org
@gustavnikolaj:registry=http://registry.npmjs.org
```

I think you'd be able to verify that the `@gustavnikolaj/migrate` package was being fetched through http and not https.

If the code paths I've changed is not being called directly from other modules, I'm confident in what I did.

**Comment by gustavnikolaj** [comment]
Oh, by the way. I fixed that embarrassing left-over console.log. I can squash the commits and force push the branch if you prefer :-)

**Comment by guybedford** [comment]
Great, thanks. I've actually been running the 0.17 development on master, and have just forced the branch back to parity with the right branch to separate out that PR. Do you think you could look at rebasing for merge?

**Comment by gustavnikolaj** [comment]
Sure, gimme a few minutes :-)

**Comment by gustavnikolaj** [comment]
I think it's in order now :-) Had to do a force push any way, so I merged the commit removing the console.log call.

File: lib/npm.js [hunk 2]
```diff
@@ -32,8 +32,13 @@ var NPMLocation = function(options, ui) {
...
 
   var npmrc = new Npmrc();
-  
-  this.registryURL = options.registry || npmrc.getRegistry() || defaultRegistry;
+
+  this.registryURL = function (scope) {
+    if (scope) {
+      return npmrc.getRegistry(scope) || options.registry || defaultRegistry;
+    }
+    return options.registry || npmrc.getRegistry() || defaultRegistry;
```
Comment (guybedford) [inline lib/npm.js]: Perhaps we can just make this:

```
  return scope && npmrc.getRegistry(scope) || options.registry || npmrc.getRegistry() || defaultRegistry;
```

Note the fallback to `npmrc.getRegistry()` for scopes.

**Comment by guybedford** [comment]
Thanks for the quick updates!

File: lib/npm.js (hunk 2 above)
Comment (gustavnikolaj) [inline lib/npm.js]: done :-)

**Comment by gustavnikolaj** [comment]
I verified that it still works.

**Pull request merged**